src/crimson/os/seastore/omap_manager/btree/btree_omap_manager.cc

```diff
@@ -106,6 +106,19 @@ BtreeOMapManager::omap_get_value(
   });
 }
 
+BtreeOMapManager::omap_set_keys_ret
+BtreeOMapManager::omap_set_keys(
+  omap_root_t &omap_root,
+  Transaction &t,
+  const std::map<std::string, ceph::bufferlist>& keys) {
+  return crimson::do_for_each(
+    keys.begin(),
```

Comment:
we cannot ensure the availability of `keys` in `crimson::do_for_each()`, as only the first element is passed to the specified function, while `crimson::do_for_each()` takes care of the remaining elements by iterating through the map, but if the first call of the function does not return a future available immediately, the recursive call of `crimson::do_for_each()` will be iterating in an already-destroyed map.